### PR TITLE
Add option to use arbitrary Sphinx HTML artifact folder

### PR DESCRIPTION
**Pull request by jakelishman**
This adds a `--sphinx-artifact-folder` (alias `--sphinx-output`) option to manually specify the root of the Sphinx HTML output for a given package.  This implies `--skip-download`.  This is convenient for local API-docs development of the targetted packages, since I can use their regular docs-build processes and then just point `gen-api` at the right place without needing to copy anything into special folders.

For example, I've locally been using Sphinx as normal in Qiskit, then running
```bash
npm run gen-api -- -p qiskit -v 1.2.0 -a /path/to/docs/_build/html
```
to have the script directly target the Sphinx files without copying them.

I made the magic `/artifact` suffix a bit more local, rather than spread between two functions for two reasons:
1. it's easier to reason about magic that's only local.
2. if the `/artifact` is _always_ appended by `zxMain` outside of the control of `prepareSphinxFolder`, it defeated the main purpose of this option.

### CODE REVIEW

File: scripts/commands/updateApiDocs.ts [hunk 1]
```diff
@@ -60,6 +61,14 @@ const readArgs = (): Arguments => {
       description:
         "Rather than downloading the artifact from Box, reuse what is already downloaded. This can save time, but it risks using an outdated version of the docs.",
     })
+    .option("sphinx-artifact-folder", {
+      alias: "sphinx-output",
+      type: "string",
+      implies: "skip-download",
+      normalize: true,
+      description:
+        "Skip downloading the the artifact from Box and instead use a given directory as the root of the Sphinx HTML output.",
```
Comment (Eric-Arellano) [inline scripts/commands/updateApiDocs.ts:70]:
```suggestion
        "Skip downloading the artifact from Box and instead use the given directory as the root of the Sphinx HTML output.",
```

File: scripts/commands/updateApiDocs.ts [hunk 2]
```diff
@@ -60,6 +61,14 @@ const readArgs = (): Arguments => {
       description:
         "Rather than downloading the artifact from Box, reuse what is already downloaded. This can save time, but it risks using an outdated version of the docs.",
     })
+    .option("sphinx-artifact-folder", {
+      alias: "sphinx-output",
```
Comment (Eric-Arellano) [inline scripts/commands/updateApiDocs.ts:65]: If you want an alias, I'd rather it be something short like `-a` for `artifact`. Right now, the two names feel quite similar and like we couldn't decide on the name

**Review by Eric-Arellano — COMMENTED** [review]
Excellent! No need to add tests. @arnaucasau will document this in https://github.com/Qiskit/documentation/pull/1336

File: scripts/commands/updateApiDocs.ts (hunk 2 above)
Comment (jakelishman) [reply scripts/commands/updateApiDocs.ts:65]: I don't really mind - this was mostly borne out of me not being to work out to have `yargs` have a command-line argument that differed from the name of the associated property in the object.  I wanted to _just_ have the command-line argument be `--sphinx-output`, but the internal details of the code want that property to be `sphinxArtifactFolder` for more consistency, which felt not-so-descriptive to me outside the context of the internal code.

Anyway, I don't mind much about the naming as long as the option exists - if you're rather I just remove the alias entirely, that's fine.

Comment (Eric-Arellano) [reply scripts/commands/updateApiDocs.ts:65]: Either remove the alias or use `-a` (or you can suggest something else). I like having the short alias to make it easier to use this script

**Review by Eric-Arellano — APPROVED** [review]
🚀

File: scripts/commands/updateApiDocs.ts (hunk 2 above)
Comment (jakelishman) [reply scripts/commands/updateApiDocs.ts]: Done in 8c6434f5d1e14843f57cd8108934d510f959bb27.

**Pull request merged**